platform/mellanox/docker-syncd-mlnx/ecmp_calc_sdk.py

```diff
@@ -0,0 +1,466 @@
+        neigh_filter.rif = 0
+        sx_neigh_filter_t_p_assign(filter_p, neigh_filter)
+        
+        rc = sx_api_router_neigh_get(handle, SX_ACCESS_CMD_GET, rif, addr, filter_p, neigh_entry_list_p, neigh_entry_cnt_p)
+        sx_check_rc(rc)
+
+        neighbor_entry = sx_neigh_get_entry_t_arr_getitem(neigh_entry_list_p, 0)
+        
+        return neighbor_entry.neigh_data.mac_addr.to_str()
+    
+    finally:
+        delete_sx_neigh_filter_t_p(filter_p)
+        delete_uint32_t_p(neigh_entry_cnt_p)
+        delete_sx_neigh_get_entry_t_arr(neigh_entry_list_p)    
+        
+        
+def sx_fdb_uc_mac_addr_get(handle, vlan_id, mac_addr):
+    """ Get UC mac entry from FDB
+    
+    Args:
+        handle (sx_api_handle_t): SDK handle
+        vlan_id (sx_vlan_id_t): VLAN id
+        mac_addr (str): mac address
+
+    Returns:
+        sx_fdb_uc_mac_addr_params_t : FDB mac entry
+    """       
+    try:
+        key = sx_fdb_uc_mac_addr_params_t()
+        key.fid_vid = vlan_id
+        key.mac_addr = ether_addr(mac_addr)
+        key.entry_type = SX_FDB_UC_AGEABLE
+        key.action = SX_FDB_ACTION_FORWARD
+        key_p = copy_sx_fdb_uc_mac_addr_params_t_p(key)
+        
+        key_filter = sx_fdb_uc_key_filter_t()
+        key_filter.filter_by_fid =  SX_FDB_KEY_FILTER_FIELD_VALID
+        key_filter.filter_by_mac_addr = SX_FDB_KEY_FILTER_FIELD_VALID
+        key_filter.filter_by_log_port = SX_FDB_KEY_FILTER_FIELD_NOT_VALID
+        key_filter.fid = vlan_id
+        key_filter.mac_addr = ether_addr(mac_addr)
+        key_filter_p = copy_sx_fdb_uc_key_filter_t_p(key_filter)
+    
+        data_cnt_p = copy_uint32_t_p(SX_FDB_MAX_GET_ENTRIES)
+        mac_list_p = new_sx_fdb_uc_mac_addr_params_t_arr(SX_FDB_MAX_GET_ENTRIES)
+        
+        rc = sx_api_fdb_uc_mac_addr_get(handle, 0, SX_ACCESS_CMD_GET_FIRST, SX_FDB_UC_AGEABLE, key_p, key_filter_p, mac_list_p, data_cnt_p)
+        sx_check_rc(rc)
+        
+        data_cnt = uint32_t_p_value(data_cnt_p)
+        assert data_cnt == 1, "Got unexpected macs amount, mac {} vlan {}".format(mac_addr, vlan_id)
+        
+        mac_entry = sx_fdb_uc_mac_addr_params_t_arr_getitem(mac_list_p, 0)
+        assert mac_entry.dest_type == SX_FDB_UC_MAC_ADDR_DEST_TYPE_LOGICAL_PORT, "Got unexpected mac entry type {}".format(mac_entry.dest_type)
+
+        return mac_entry
+
+    finally:
+        delete_sx_fdb_uc_key_filter_t_p(key_filter_p)
+        delete_sx_fdb_uc_mac_addr_params_t_p(key_p)
+        delete_sx_fdb_uc_mac_addr_params_t_arr(mac_list_p)
+        delete_uint32_t_p(data_cnt_p)       
+        
+def sx_lag_port_group_get(handle, lag_id):
+    """ Get LAG members
+    
+    Args:
+        handle (sx_api_handle_t): SDK handle
+        lag_id (sx_port_log_id_t): LAG id
+
+    Returns:
+        list : list of LAG members logical indices
+    """      
+    try:
+        lag_members = []
+        port_count_p = copy_uint32_t_p(0)
```

Comment:
@liorghub why not `new_uint32_t_p`? What is the difference?

Reply:
Calling copy_uint32_t_p is equivalent to new_uint32_t_p + uint32_t_p_assign. (please have a look at sonic-sairedis/pyext/py2/pysairedis_wrap.cpp)
However, to be consistent, I changed it to new + assign.